include/isoengine/game_object.h

```diff
@@ -1,27 +1,32 @@
 #ifndef GAMEOBJECT_H
 #define GAMEOBJECT_H
 
-#include "isoengine/common/hashedstring.h"
-#include "isoengine/events/commandqueue.h"
+#include "isoengine/support/hashed_string.h"
+#include "isoengine/events/command_queue.h"
 #include "isoengine/math/transform.h"
 #include "isoengine/render/animator.h"
-#include "scenenode.h"
+#include "isoengine/render/scene_node.h"
 #include <initializer_list>
 
-namespace iso
+namespace iso::physics
 {
 
 class CollisionDetector;
 
-class GameObject : public Animator, public SceneNode
```

Comment:
Ne legyen a GameObject inkább a SceneNode-dal egyben? Szerintem a GameObject inkább a SceneNode egy speciális típusa, ahogy későbbiekben más, akár generált objektumok is lehetnek azok. Szerintem a többi render osztállyal együtt nem feltétlenül jók, mert azok tényleg csak kirajzolásról szólnak, nem "engine" specfikusok. És ha scenenode/renderscene + gameobject együtt lesznek, akkor elég lesz csak az engine.h-t kívülre rakni. Mit gondolsz?